packages/office-ui-fabric-react/src/components/DocumentCard/DocumentCard.types.ts

```diff
@@ -63,257 +48,61 @@ export interface IDocumentCardProps extends IBaseProps<IDocumentCard> {
-   */
-  locationHref?: string;
-
-  /**
-   * Function to call when the location is clicked.
-   */
-  onClick?: (ev?: React.MouseEvent<HTMLElement>) => void;
-
-  /**
-   * Aria label for the link to the document location.
-   */
-  ariaLabel?: string;
-}
-
-export interface IDocumentCardActivityProps extends React.ClassAttributes<DocumentCardActivity> {
-  /**
-   * Gets the component ref.
-   */
-  componentRef?: IRefObject<{}>;
-
-  /**
-   * Describes the activity that has taken place, such as "Created Feb 23, 2016".
-   */
-  activity: string;
-
-  /**
-   * One or more people who are involved in this activity.
-   */
-  people: IDocumentCardActivityPerson[];
+  className?: string;
 }
 
-export interface IDocumentCardActivityPerson {
-  /**
-   * The name of the person.
-   */
-  name: string;
-
-  /**
-   * Path to the profile photo of the person.
-   */
-  profileImageSrc: string;
-
-  /**
-   * The user's initials to display in the profile photo area when there is no image.
-   */
-  initials?: string;
-
+export enum DocumentCardType {
   /**
-   * Whether initials are calculated for phone numbers and number sequences.
-   * Example: Set property to true to get initials for project names consisting of numbers only.
-   * @defaultvalue false
+   * Standard DocumentCard.
    */
-  allowPhoneInitials?: boolean;
-
+  normal = 0,
   /**
-   * The background color when the user's initials are displayed.
-   * @defaultvalue PersonaInitialsColor.blue
+   * Compact layout. Displays the preview beside the details, rather than above.
    */
-  initialsColor?: PersonaInitialsColor;
+  compact = 1
 }
 
-export interface IDocumentCardActionsProps extends React.ClassAttributes<DocumentCardActions> {
-  /**
-   * Gets the component ref.
-   */
-  componentRef?: IRefObject<{}>;
-
+export interface IDocumentCardStyleProps {
   /**
-   * The actions available for this document.
+   * Accept theme prop.
    */
-  actions: IButtonProps[];
+  theme: ITheme;
 
   /**
-   * The number of views this document has received.
+   * Optional override class name
    */
-  views?: Number;
-}
+  className?: string;
 
-export interface IDocumentCardLogoProps extends React.ClassAttributes<DocumentCardLogo> {
-  /**
-   * Gets the component ref.
-   */
-  componentRef?: IRefObject<{}>;
   /**
-   * Describes DocumentCard Logo badge.
+   * True when the card has a click action.
    */
-  logoIcon: string;
+  actionable?: boolean;
 
   /**
-   * Describe Logo name, optional.
+   * Compact variant of the card.
    */
-  logoName?: string;
+  compact?: boolean;
 }
 
-export interface IDocumentCardStatusProps extends React.ClassAttributes<DocumentCardStatus> {
-  /**
-   * Gets the component ref.
-   */
-  componentRef?: IRefObject<{}>;
-  /**
-   * Describes DocumentCard status icon.
-   */
-  statusIcon?: string;
-
-  /**
-   * Describe status information. Required field.
-   */
-  status: string;
+export interface IDocumentCardStyles {
+  root: IStyle;
```

Comment:
nit: Would be nice to have the interface itself to be documented and each section with a short description of what it represents. I just hate to see adding new undocumented APIs.

Same goes for all components 🙄